cmd/rule-evaluator/main.go

```diff
@@ -542,8 +583,13 @@ func sendAlerts(s *notifier.Manager, projectID string, externalURL *url.URL) rul
 			}
 			if externalURL != nil {
 				if externalURL.String() == googleCloudBaseURL.String() {
-					// If it's a GCM link (default), create the full URL for the alert.
-					a.GeneratorURL = googleCloudLink(projectID, expr, alert.FiredAt, alert.FiredAt.Add(-time.Hour)).String()
+					// Project ID is empty when the rule-evaluator is instantiated, before config-reloader runs.
+					if projectID == "" {
+						a.GeneratorURL = ""
+					} else {
+						// If it's a GCM link (default), create the full URL for the alert.
+						a.GeneratorURL = googleCloudLink(projectID, expr, alert.FiredAt, alert.FiredAt.Add(-time.Hour)).String()
+					}
```

Comment:
So this `// Project ID is empty when the rule-evaluator is instantiated, before config-reloader runs.` comment is likely wrong. This is `sendAlert` that is unlikely to be called when no rules or alerts are set. Perhaps it's fine to ignore this check here? 

Or... if `googleCloudLink` has empty project ID we could simply set it to "<ProjectID>" so it's clear we don't know, but I am sceptical we would ever hit this here.

AND if you really want this check, maybe this is more readable: 

```suggestion
					if projectID != "" {
						// If it's a GCM link (default), and we have a projectID set, create the full URL for the alert.
						a.GeneratorURL = googleCloudLink(projectID, expr, alert.FiredAt, alert.FiredAt.Add(-time.Hour)).String()
					}
```

Reply:
Thanks! I'd like to keep this check because you need a project ID for a valid link, so I made the readability change.